scripts/cardano-node.sh

```diff
@@ -21,21 +20,15 @@ node_config_files=(
 trap 'kill $cardano_node_pid; exit' INT TERM QUIT ERR EXIT
 
 set -x
-mkdir -p "$NODE_BIN_DIR"
-# Download cardano-node binary. The -nc option avoids downloading if the file
-# already exists locally.
-wget -nc https://hydra.iohk.io/build/17088436/download/1/$node_zipped -P "$NODE_BIN_DIR"
-
-tar zxvf "$NODE_BIN_DIR"/$node_zipped -C "$NODE_BIN_DIR"
 
 mkdir -p "$NODE_DIR"
-# Download config files
-for x in "${node_config_files[@]}"; do
-  wget -nc https://hydra.iohk.io/build/16607585/download/1/"$x" -P "$NODE_DIR"
-done
+# # Download config files
+# for x in "${node_config_files[@]}"; do
+#   wget -nc https://hydra.iohk.io/build/16607585/download/1/"$x" -P "$NODE_DIR"
+# done
 
 # Launch node
-"$NODE_BIN_DIR"/cardano-node run \
+cardano-node run \
```

Comment:
I need a little help with this.
The hydra links are now broken, and they need to be replaced with moritz's.

However, I see here that we are downloading the 1.35.2 binary every time.
This is not necessary, as we have 1.35.4 in PATH from the shell.

However we still need to get the node_config_files.

Where can they be found? Must we rely on a hard-coded hydra path?
Aren't they included maybe in the cardano-node derivation? Maybe in share or etc?
Are they part of the cardano-node repo?
Can they instead be copy-pasted in this script folder?
Can someone take on fixing this script and making it work locally, and then I can make sure that it can run as a GHA?

@michaelpj @zliu41

Reply:
@zeme-iohk I'm trying to rebase the code on top of plutus-apps HEAD and make it work. This could take a few days.

I don't know the answer to your questions, in particular how to use a non-hardcoded hydra path.

> downloading the 1.35.2 binary every time.

The job needs to run on the same machine every time (since we don't want to re-sync the node every time), and if it does then it shouldn't need to download it every time.

Reply:
Ok, we'll have to use a hard-coded hydra path for the config then, but again, `cardano-node` is already in PATH so there should be no need to download it (unless you really need the 1.35.2 binary, and not the 1.35.4)

Reply:
@zeme-iohk It's working for me locally now:
```
git checkout zliu41/dump-1.35.4

AWS_ACCESS_KEY_ID=<...> \
  AWS_SECRET_ACCESS_KEY=<...> \
  AWS_DEFAULT_REGION=us-east-1 \
  AWS_ENDPOINT_URL=https://s3.devx.iog.io \
  NODE_DIR=$HOME/mainnet \
  S3_DUMP_DIR=s3://plutus/mainnet-script-dump-1-35-4/ \
  LOCAL_DUMP_DIR=$HOME/mainnet-script-dump \
  ./scripts/create-dump.sh
```

It uses the `cardano-node` on the PATH. For the config files, it downloads them from `book.world.dev.cardano.org`, as instructed [here](https://developers.cardano.org/docs/get-started/running-cardano/).

The job used to run daily on buildkite for 3 hours. It would be great to do the same with GHA.

Reply:
There are no new commits to the PR... am I missing something? 🤔

Reply:
Also, are we using `mainnet-script-dump-1-35-4` or `mainnet-script-dump`?

Reply:
There's no PR. It is not mergeable until plutus-app `main` branch starts using a newer version of cardano-node and plutus.

Let's use `mainnet-script-dump-1-35-4` to make it clear the node version used to generate the files.

Reply:
I.e., the implementation of the dump job is in branch `zliu41/dump-1.35.4`, which is the branch the GH action should use.

Reply:
Got it, I'll check it out. thanks!